examples/8b_benchmark_read1D2D_parallel.cpp

```diff
@@ -0,0 +1,659 @@
+     *
+     * @param tag      item name to measure
+     * @param rank     MPI rank
+     */
+
+  void Display(const char*  tag){
+    if (0 == m_Name.size())
+      return;
+
+    if (m_Rank > 0)
+      return;
+
+    std::cout<<" memory at:  "<<tag;
+    std::ifstream input(m_Name.c_str());
+
+    if (input.is_open()) {
+      for (std::string line; getline(input, line);)
+        {
+          if (line.find("VmRSS") == 0)
+        std::cout<<line<<" ";
+          if (line.find("VmSize") == 0)
+        std::cout<<line<<" ";
+          if (line.find("VmSwap") == 0)
+        std::cout<<line;
+        }
+      std::cout<<std::endl;
+      input.close();
+    }
+  }
+private:
+  int m_Rank;
+  std::string  m_Name;
+};
+
+
+/** The Timer class for profiling purpose
+ *
+ *  Simple Timer that measures time consumption btw constucture and destructor
+ *  Reports at rank 0 at the console, for immediate convenience
+ */
+class Timer
+{
+public:
+    /**
+     *
+     * Simple Timer
+     *
+     * @param tag      item name to measure
+     * @param rank     MPI rank
+     */
+    Timer(const char* tag, int rank) {
+        m_Tag = tag;
+        m_Rank = rank;
+        m_Start = std::chrono::system_clock::now();
+    MemoryProfiler (rank, tag);
+    }
+    ~Timer() {
+        std::string tt = "~"+m_Tag;
+        MemoryProfiler (m_Rank, tt.c_str());
+        m_End = std::chrono::system_clock::now();
+
+        double millis = std::chrono::duration_cast< std::chrono::milliseconds >( m_End - m_Start ).count();
+        double secs = millis/1000.0;
+        if( m_Rank > 0 )
+          return;
+
+        std::cout << "  [" << m_Tag << "] took:" << secs << " seconds\n";
+        std::cout<<"     " << m_Tag <<"  From ProgStart in seconds "<<
+      std::chrono::duration_cast<std::chrono::milliseconds>(m_End - m_ProgStart).count()/1000.0<<std::endl;
+
+    }
+private:
+    std::chrono::time_point<std::chrono::system_clock> m_Start;
+    std::chrono::time_point<std::chrono::system_clock> m_End;
+
+    std::string m_Tag;
+    int m_Rank = 0;
+};
+
+
+/** ... createData
+ *      generate a shared ptr of given size  with given type & default value
+ *
+ * @param T   ......... data type
+ * @param size ... .... data size
+ * @param val  ........ data value by default
+ *
+ */
+
+template<typename T>
+std::shared_ptr< T > createData(const unsigned long& size,  const T& val )
+  {
+    auto E = std::shared_ptr< T > {
+      new T[size], []( T * d ) {delete[] d;}
+    };
+
+    for(unsigned long  i = 0ul; i < size; i++ )
+      {
+        E.get()[i] = val;
+      }
+    return E;
+  }
+
+/** Find supported backends
+ *  (looking for ADIOS2 or H5)
+ *
+ */
+std::vector<std::string> getBackends() {
+    std::vector<std::string> res;
+#if openPMD_HAVE_ADIOS2
+    if( auxiliary::getEnvString( "OPENPMD_BP_BACKEND", "NOT_SET" ) != "ADIOS1" )
+        res.emplace_back(".bp");
+#endif
+
+#if openPMD_HAVE_HDF5
+    res.emplace_back(".h5");
+#endif
+    return res;
+}
```

Comment:
You can use `openPMD::getFileExtensions()` if you update against `dev` (added in #768)

Reply:
getFileExtensions() is a useful addition. And since I don't want to include json files,  will leave it as is.